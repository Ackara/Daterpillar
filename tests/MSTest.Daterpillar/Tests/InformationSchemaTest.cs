using Acklann.Daterpillar.Migration;
using Acklann.Daterpillar.Scripting;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MSTest.Daterpillar.Tests
{
    [TestClass]
    [DeploymentItem(FName.X86)]  
    [DeploymentItem(FName.Samples)]
    public class InformationSchemaTest
    {
        [TestMethod]
        public void FetchSchema_should_return_schema_object_based_on_an_existing_mysql_database()
        {
            using (var connection = ConnectionFactory.CreateMySQLConnection(DBNAME))
            {
                connection.UseSchema(FName.scriptingTest_partial_schemaXML);
                var sut = new MySQLInformationSchema(connection);
                var result = sut.FetchSchema(DBNAME);
                var settings = new SqlScriptBuilderSettings()
                {
                    AppendCreateSchemaCommand = false,
                    IgnoreComments = true,
                    IgnoreScripts = true,
                    ShowHeader = false,
                };

                var mysql = new MySQLScriptBuilder(settings)
                    .Append(result)
                    .GetContent();

                Approvals.Verify(mysql);
            }
        }

        [TestMethod]
        public void FetchSchema_should_return_schema_object_based_on_an_existing_mssql_database()
        {
            using (var connection = ConnectionFactory.CreateMSSQLConnection(DBNAME))
            {
                connection.UseSchema(FName.scriptingTest_partial_schemaXML);
                var sut = new MSSQLInformationSchema(connection);
                var result = sut.FetchSchema(DBNAME);
                var settings = new SqlScriptBuilderSettings()
                {
                    AppendCreateSchemaCommand = false,
                    IgnoreComments = true,
                    IgnoreScripts = true,
                    ShowHeader = false,
                };

                var mssql = new MSSQLScriptBuilder(settings)
                    .Append(result)
                    .GetContent();

                Approvals.Verify(mssql);
            }
        }

        [TestMethod]
        public void FetchSchema_should_return_schema_object_based_on_an_existing_sqlite_database()
        {
            using (var connection = ConnectionFactory.CreateSQLiteConnection(DBNAME))
            {
                connection.UseSchema(FName.scriptingTest_partial_schemaXML);
                var sut = new SQLiteInformationSchema(connection);
                var result = sut.FetchSchema(DBNAME);
                var settings = new SqlScriptBuilderSettings()
                {
                    AppendCreateSchemaCommand = false,
                    IgnoreComments = true,
                    IgnoreScripts = true,
                    ShowHeader = false,
                };

                var mssql = new SQLiteScriptBuilder(settings)
                    .Append(result)
                    .GetContent();

                Approvals.Verify(mssql);
            }
        }

        private const string DBNAME = "dtpl_info_mockDb";
    }
}