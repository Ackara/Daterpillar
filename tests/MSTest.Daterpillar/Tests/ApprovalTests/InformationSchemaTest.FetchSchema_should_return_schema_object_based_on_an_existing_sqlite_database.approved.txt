CREATE TABLE IF NOT EXISTS [card_type]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(32) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [card_type_Name] ON [card_type] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [card_type_Id] ON [card_type] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [attribute]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(16) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [attribute_Name] ON [attribute] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [attribute_Id] ON [attribute] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [card_icon]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(16) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [card_icon_Name] ON [card_icon] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [card_icon_Id] ON [card_icon] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [monster_type]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(32) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [monster_type_Name] ON [monster_type] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [monster_type_Id] ON [monster_type] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [ability]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(32) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [ability_Name] ON [ability] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [ability_Id] ON [ability] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [card]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(128) NOT NULL,
	[Text] TEXT NOT NULL,
	[Pendulum_Effect] TEXT NOT NULL,
	[Level] INTEGER NOT NULL DEFAULT ''0'',
	[Atk] INTEGER NOT NULL,
	[Def] INTEGER NOT NULL,
	[Card_Type_Id] INTEGER NOT NULL,
	[Attribute_Id] INTEGER NOT NULL,
	[Card_Icon_Id] INTEGER NOT NULL,
	[Monster_Type_Id] INTEGER NOT NULL,
	[Ability_Id] INTEGER NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [card_Name] ON [card] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [card_Id] ON [card] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [card_extras]
(
	[Card_Id] INTEGER NOT NULL,
	[Rulings] TEXT,
	[Tips] TEXT,
	[Trivia] TEXT,
	[Passcode] INTEGER NOT NULL
);

CREATE INDEX IF NOT EXISTS [card_extras_Card_Id] ON [card_extras] ([Card_Id] ASC);

CREATE TABLE IF NOT EXISTS [pack]
(
	[Id] INTEGER NOT NULL,
	[Acronym] VARCHAR(32) NOT NULL,
	[Name] VARCHAR(128) NOT NULL,
	[Description] TEXT NOT NULL,
	[Release_Date] DATE NOT NULL,
	[Size] INTEGER NOT NULL,
	[Konami_Id] INTEGER NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [pack_Name] ON [pack] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [pack_Id] ON [pack] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [rarity]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(32) NOT NULL,
	[Code] VARCHAR(4) NOT NULL,
	[Value] INTEGER NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [rarity_Name] ON [rarity] ([Name] ASC);
CREATE INDEX IF NOT EXISTS [rarity_Id] ON [rarity] ([Id] ASC);

CREATE TABLE IF NOT EXISTS [card_number]
(
	[Pack_Acronym] VARCHAR(32) NOT NULL,
	[Number] INTEGER NOT NULL,
	[Rarity_Id] INTEGER NOT NULL,
	[Art_Index] INTEGER NOT NULL,
	[Card_Id] INTEGER NOT NULL,
	[Pack_Id] INTEGER NOT NULL
);

CREATE INDEX IF NOT EXISTS [card_number_Pack_Acronym_and_Number_and_Rarity_Id_and_Art_Index] ON [card_number] ([Pack_Acronym] ASC, [Number] ASC, [Rarity_Id] ASC, [Art_Index] ASC);

CREATE TABLE IF NOT EXISTS [effect]
(
	[Id] INTEGER NOT NULL,
	[Name] VARCHAR(32) NOT NULL
);

CREATE UNIQUE INDEX IF NOT EXISTS [effect_Name] ON [effect] ([Name] ASC);

